framework/test/unit/modules/token/commands/transfer.spec.ts

```diff
@@ -160,277 +201,268 @@ describe.skip('Transfer command', () => {
-						amount,
-						recipientAddress,
+						tokenID: Buffer.from('0000000100000000', 'hex'),
+						amount: BigInt(100000000),
+						recipientAddress: utils.getRandomBytes(20),
 						data: '1'.repeat(64),
+						accountInitializationFee: BigInt(USER_SUBSTORE_INITIALIZATION_FEE),
 					}),
 					signatures: [utils.getRandomBytes(64)],
 				}),
 			});
-			await expect(
-				command.execute(context.createCommandExecuteContext(transferParamsSchema)),
-			).rejects.toThrow('does not satisfy min balance requirement');
+			const result = await command.verify(context.createCommandVerifyContext(transferParamsSchema));
+			expect(result.status).toEqual(VerifyStatus.FAIL);
+			expect(result.error?.message).toInclude('Insufficient balance.');
 		});
 
-		// TODO: Fix in https://github.com/LiskHQ/lisk-sdk/issues/7573
-		it.skip('should resolve when recipient does not exist but amount is greater than minBalance', async () => {
-			const recipientAddress = utils.getRandomBytes(20);
+		it('should pass if balance for the provided tokenID is sufficient', async () => {
 			const amount = BigInt(100000000);
+
+			command['_method'].getAvailableBalance = jest.fn().mockResolvedValue(amount);
+
 			const context = createTransactionContext({
-				stateStore,
 				transaction: new Transaction({
 					module: 'token',
 					command: 'transfer',
-					fee: BigInt(0),
+					fee: BigInt(5000000),
 					nonce: BigInt(0),
-					senderPublicKey: sender.publicKey,
+					senderPublicKey: utils.getRandomBytes(32),
 					params: codec.encode(transferParamsSchema, {
-						tokenID,
+						tokenID: Buffer.from('0000000100000000', 'hex'),
 						amount,
-						recipientAddress,
+						recipientAddress: utils.getRandomBytes(20),
 						data: '1'.repeat(64),
+						accountInitializationFee: BigInt(USER_SUBSTORE_INITIALIZATION_FEE),
 					}),
 					signatures: [utils.getRandomBytes(64)],
 				}),
 			});
-			await expect(
-				command.execute(context.createCommandExecuteContext(transferParamsSchema)),
-			).resolves.toBeUndefined();
-
-			// Recipient should receive amount - min balance if not exist
-			const userStore = tokenModule.stores.get(UserStore);
-			const result = await userStore.get(
-				context.createCommandExecuteContext(transferParamsSchema),
-				userStore.getKey(recipientAddress, localTokenID),
-			);
-			expect(result.availableBalance).toEqual(amount - MIN_BALANCE);
-
-			// Min balance is burnt
-			const supplyStore = tokenModule.stores.get(SupplyStore);
-			const supply = await supplyStore.get(
-				context.createCommandExecuteContext(transferParamsSchema),
-				localTokenID.slice(4),
-			);
-			expect(supply.totalSupply).toEqual(totalSupply - MIN_BALANCE);
+			const result = await command.verify(context.createCommandVerifyContext(transferParamsSchema));
+			expect(result.status).toEqual(VerifyStatus.OK);
 		});
+	});
 
-		it('should resolve and not burn supply when tokenID is not native and recipient does not exist', async () => {
+	describe('execute', () => {
```

Comment:
Please add cases
1. sender has enough balance for initialization fee but not for amount
2. sender and recipient are the same